attestation-agent/deps/crypto/src/rust/ec.rs

```diff
@@ -0,0 +1,177 @@
+    }
+
+    pub fn unwrap_key(
+        &self,
+        encrypted_key: Vec<u8>,
+        epk_x: Vec<u8>,
+        epk_y: Vec<u8>,
+        wrapping_algorithm: KeyWrapAlgorithm,
+    ) -> Result<Vec<u8>> {
+        match wrapping_algorithm {
+            KeyWrapAlgorithm::EcdhEsA256Kw => {
+                let secret_key = self.secret_key();
+                let epk_x: [u8; 32] = epk_x
+                    .try_into()
+                    .map_err(|_| anyhow!("invalid bytes length of coordinates X"))?;
+                let epk_y: [u8; 32] = epk_y
+                    .try_into()
+                    .map_err(|_| anyhow!("invalid bytes length of coordinates Y"))?;
+                let epk_point = EncodedPoint::from_affine_coordinates(
+                    &GenericArray::from(epk_x),
+                    &GenericArray::from(epk_y),
+                    false,
+                );
+                let public_key = P256PublicKey::from_encoded_point(&epk_point)
+                    .into_option()
+                    .ok_or(anyhow!("invalid public key"))?;
+
+                let z = diffie_hellman(secret_key.to_nonzero_scalar(), public_key.as_affine())
+                    .raw_secret_bytes()
+                    .to_vec();
+
+                let mut key_derivation_materials = Vec::new();
+                let algorithm_str = KeyWrapAlgorithm::EcdhEsA256Kw.as_ref();
+                key_derivation_materials
+                    .extend_from_slice(&(algorithm_str.len() as u32).to_be_bytes());
+                key_derivation_materials.extend_from_slice(algorithm_str.as_bytes());
+                key_derivation_materials.extend_from_slice(&(0_u32).to_be_bytes());
+                key_derivation_materials.extend_from_slice(&(0_u32).to_be_bytes());
+                key_derivation_materials.extend_from_slice(&AES_GCM_256_KEY_BITS.to_be_bytes());
+                let mut unwrapping_key = vec![0; 32];
+                concat_kdf::derive_key_into::<rsa::sha2::Sha256>(
+                    &z,
+                    &key_derivation_materials,
+                    &mut unwrapping_key,
+                )
+                .map_err(|e| anyhow!("failed to do concat KDF: {e:?}"))?;
+                let unwrapping_key: [u8; 32] = unwrapping_key
+                    .try_into()
+                    .map_err(|_| anyhow!("invalid bytes length of AES wrapping key"))?;
+                let unwrapping_key: KekAes256 = Kek::new(&GenericArray::from(unwrapping_key));
+                let mut decrypted_key = vec![0; encrypted_key.len() - 8];
+                unwrapping_key
+                    .unwrap(&encrypted_key, &mut decrypted_key)
+                    .map_err(|e| anyhow!("failed to unwrap key: {e:?}"))?;
+
+                Ok(decrypted_key)
+            }
+        }
+    }
+}
+
+#[derive(Clone, Debug)]
+pub struct P256EcKeyPair {
+    secret_key: SecretKey,
+    public_key: P256PublicKey,
+}
+
+impl Default for P256EcKeyPair {
+    fn default() -> Self {
```

Comment:
The openssl version has `new` instead of `default` any reason why they aren't the same?